cpp/daal/src/externals/service_service_mkl.h

```diff
@@ -37,49 +37,106 @@ namespace mkl
 {
 struct MklService
 {
-    static void * serv_malloc(size_t size, size_t alignment) { return fpk_serv_malloc(size, alignment); }
+    static void * serv_malloc(size_t size, size_t alignment) { return MKL_malloc(size, alignment); }
 
-    static void serv_free(void * ptr) { fpk_serv_free(ptr); }
+    static void serv_free(void * ptr) { MKL_free(ptr); }
 
-    static void serv_free_buffers() { fpk_serv_free_buffers(); }
+    static void serv_free_buffers() { MKL_Free_Buffers(); }
 
     static int serv_memcpy_s(void * dest, size_t destSize, const void * src, size_t srcSize)
     {
-        return fpk_serv_memcpy_s(dest, destSize, src, srcSize);
+        if (destSize < srcSize) return static_cast<int>(ENOMEM);
+        memcpy(dest, src, srcSize);
+        return 0;
+        // TODO: safe funtion
+        // return memcpy_s(dest, destSize, src, srcSize);
     }
 
-    static int serv_memmove_s(void * dest, size_t destSize, const void * src, size_t smax) { return fpk_serv_memmove_s(dest, destSize, src, smax); }
-
-    static int serv_get_ht() { return fpk_serv_get_ht(); }
+    static int serv_memmove_s(void * dest, size_t destSize, const void * src, size_t smax)
+    {
+        if (destSize < smax) return static_cast<int>(ENOMEM);
+        memmove(dest, src, smax);
+        return 0;
+        // TODO: safe funtion
+        // return memmove_s(dest, destSize, src, smax);
+    }
 
-    static int serv_get_ncpus() { return fpk_serv_get_ncpus(); }
+    static int serv_get_ht()
+    {
+        // TODO: real detection of hypertheading
```

Comment:
The implementation should be similar to what is done in this test
https://github.com/oneapi-src/oneDAL/blob/main/cpp/oneapi/dal/test/global_context.cpp#L97
Need to check the bits [23:16] of ebx register after calling `cpuid(0x1, registers);`. I can help with the actual implementation.